Client/Frontend/Browser/TabLocationView.swift

```diff
@@ -430,4 +434,28 @@ private class DisplayTextField: UITextField {
     fileprivate override var canBecomeFirstResponder: Bool {
         return false
     }
+    
+    override func textRect(forBounds bounds: CGRect) -> CGRect {
+        var rect: CGRect = super.textRect(forBounds: bounds)
+        
+        if let size: CGSize = (self.text as NSString?)?.size(withAttributes: [.font: self.font!]) {
+            if size.width > self.bounds.width {
+                rect.origin.x = rect.origin.x - (size.width - self.bounds.width)
+                rect.size.width = rect.size.width + (size.width - self.bounds.width)
+            }
+        }
+        return rect
+    }
+}
+
+extension String {
+    func etldValue(parser: DomainParser?) -> String {
+        guard parser != nil else {
```

Comment:
You can unwrap `parser` with `guard let parser = parser else { return self }`

Reply:
Done